pages/portfolio.js

```diff
@@ -0,0 +1,155 @@
+/* global BACKEND_URL, fetch */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Layout from '../components/layout/main';
+import Background from '../components/content/background';
+
+import Works from '../components/portfolio/works';
+
+import { works } from '../main.config';
+
+const transformateCategories = (choosenCategory, existCategories) => {
+  const categories = [];
+
+  existCategories.forEach((existCategory) => {
+    choosenCategory.forEach((category) => {
+      if (category.toLowerCase() === existCategory.toLowerCase() && categories.indexOf(existCategory) === -1) {
+        categories.push(existCategory);
+      }
+    });
+  });
+
+  return categories.length ? categories : existCategories;
+}
+
+export default class Portfolio extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      ...this.getFilterList.call(this)
+    };
+
+    this.worksCountFor = this.worksCountFor.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.showAllWorks = this.showAllWorks.bind(this);
+    this.disabledBtn = this.disabledBtn.bind(this);
+  }
+
+  componentDidMount() {
+    const subnavigation = document.querySelector('.navigation-item.current > .subnavigation');
+    subnavigation.style.display = 'none';
+    subnavigation.parentElement.classList.add('is-link');
+  }
+
+  getFilterList() {
+    const { url } = this.props;
+    const filterList = [];
+    const choosenCategory = url.query.choosen;
+
+    works.forEach((work) => {
+      work.category.main.forEach((category) => {
+        if (filterList.indexOf(category) === -1) {
+          filterList.push(category);
+        }
+      });
+    });
```

Comment:
@dmytropanontko , could you use `filter` here?

Reply:
@epaminond I haven't idea how rewrite this part of with filter

Reply:
@dmytropanontko , let's try it this way:
```js
const categories = works
  .map(work => work.category.main)
  .reduce((a, b) => a.concat(Array.isArray(b) ? flatten(b) : b), []) // flatten
const uniqCategories = [...new Set(categories)]
```